lib/ecto/adapters/sql.ex

```diff
@@ -245,6 +245,31 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
+  @doc """
+  TODO: adapter's specific details should be documented here?
+
+  ## Examples
+
+      iex> Ecto.Adapters.SQL.explain(:all, repo, Post)
+      "Seq Scan on posts p0  (cost=0.00..12.12 rows=1 width=443)"
+
+  """
+  @spec explain(Ecto.Repo.t, :all | :update_all | :delete_all, Ecto.Queryable.t, Keyword.t) :: String.t
+  def explain(repo, operation, queryable, opts \\ []) do
+    {prepared, params} = to_sql(operation, repo, queryable)
+    %{sql: sql} = adapter_meta = Ecto.Adapter.lookup_meta(repo)
+
+    {:ok, explain} =
+      transaction(adapter_meta, [], fn ->
+        query = sql.explain_query(prepared, opts)
+        {:ok, %{rows: rows}} = sql_call(adapter_meta, :query, [query], params, [])
```

Comment:
A pattern match for a error clause is required here. The question is if it should raise or if `explain` could just return `nil` if the `sql_call` call fails.

Reply:
Can you use the high level `query(...)` API instead?